src/main/java/uk/antiperson/stackmob/entity/EntityManager.java

```diff
@@ -1,25 +1,103 @@
 package uk.antiperson.stackmob.entity;
 
+import org.bukkit.Bukkit;
+import org.bukkit.Chunk;
+import org.bukkit.World;
+import org.bukkit.entity.Entity;
 import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Mob;
 import org.bukkit.persistence.PersistentDataType;
 import uk.antiperson.stackmob.StackMob;
 
+import java.util.Collection;
+import java.util.HashSet;
+
 public class EntityManager {
 
     private final StackMob sm;
+    private final HashSet<StackEntity> stackEntities;
     public EntityManager(StackMob sm) {
         this.sm = sm;
+        stackEntities = new HashSet<>();
     }
 
     public boolean isStackedEntity(LivingEntity entity) {
         return entity.getPersistentDataContainer().has(sm.getStackKey(), PersistentDataType.INTEGER);
     }
 
+    public HashSet<StackEntity> getStackEntities() {
+        return stackEntities;
+    }
+
     public StackEntity getStackEntity(LivingEntity entity) {
-        return new StackEntity(sm, entity);
+        for (StackEntity stackEntity : stackEntities) {
+            if (stackEntity.getEntity().getUniqueId().equals(entity.getUniqueId())) {
+                return stackEntity;
+            }
+        }
+        return null;
+    }
+
+    public void registerAllEntities() {
+        for (World world : Bukkit.getWorlds()) {
+            for (Chunk chunk : world.getLoadedChunks()) {
+                registerStackedEntities(chunk);
+            }
+        }
+    }
+
+    public void unregisterAllEntities() {
```

Comment:
If we got rid of the chunk listener, is this still required?

Reply:
Not really.